pennylane/devices/default_qutrit.py

```diff
@@ -0,0 +1,258 @@
+
+    # TODO: Update list of operations and observables once more are added
+    operations = {
+        "QutritUnitary",
+    }
+
+    observables = {
+        "Identity",
+    }
+
+    def __init__(
+        self,
+        wires,
+        *,
+        r_dtype=np.float64,
+        c_dtype=np.complex128,
+        shots=None,
+        analytic=None,
+    ):
+        super().__init__(wires, shots, r_dtype=r_dtype, c_dtype=c_dtype, analytic=analytic)
+        self._debugger = None
+
+        # Create the initial state. Internally, we store the
+        # state as an array of dimension [3]*wires.
+        self._state = self._create_basis_state(0)
+        self._pre_rotated_state = self._state
+
+        # TODO: Add operations
+        self._apply_ops = {
+            # All operations that can be applied on the `default.qutrit` device by directly
+            # manipulating the internal state array will be included in this set
+        }
+
+    @functools.lru_cache()
+    def map_wires(self, wires):
+        # temporarily overwrite this method to bypass
+        # wire map that produces Wires objects
+        try:
+            mapped_wires = [self.wire_map[w] for w in wires]
+        except KeyError as e:
+            raise WireError(
+                f"Did not find some of the wires {wires.labels} on device with wires {self.wires.labels}."
+            ) from e
+
+        return mapped_wires
+
+    def define_wire_map(self, wires):
+        # temporarily overwrite this method to bypass
+        # wire map that produces Wires objects
+        consecutive_wires = range(self.num_wires)
+        wire_map = zip(wires, consecutive_wires)
+        return dict(wire_map)
+
+    def apply(self, operations, rotations=None, **kwargs):  # pylint: disable=arguments-differ
+        rotations = rotations or []
+
+        # apply the circuit operations
+        for i, operation in enumerate(operations):  # pylint: disable=unused-variable
+            self._state = self._apply_operation(self._state, operation)
+
+        # store the pre-rotated state
+        self._pre_rotated_state = self._state
+
+        # apply the circuit rotations
+        for operation in rotations:
+            self._state = self._apply_operation(self._state, operation)
+
+    def _apply_operation(self, state, operation):
+        """Applies operations to the input state.
+
+        Args:
+            state (array[complex]): input state
+            operation (~.Operation): operation to apply on the device
+
+        Returns:
+            array[complex]: output state
+        """
+        if operation.base_name == "Identity":
+            return state
+        wires = operation.wires
+
+        if operation.base_name in self._apply_ops:
+            axes = self.wires.indices(wires)
+            return self._apply_ops[operation.base_name](state, axes, inverse=operation.inverse)
+
+        matrix = self._asarray(self._get_unitary_matrix(operation), dtype=self.C_DTYPE)
+
```

Comment:
Will the check for the op being diagonal be added in a future PR? I assume that only applies to the "normal" (i.e., non-`QutritUnitary`operations?

Reply:
Yes. There will be a PR for adding optimizations such as optimizing for diagonal matrices, using `einsum` instead of `tensordot` for small operations, etc. Currently, the focus is to just get the device and operations up and running :)